### ISSUE TITLE
Deployments Not Updating / Displaying Properly

### ISSUE THREAD

**Issue by zeldafreak6245** [sw5e-foundry/sw5e#708]
**Describe the bug**
Deployments seem to be non-functional in general. None of their actions or abilities appear to be listed in the "actions" portion of this tab.

**To Reproduce**
1. Open a PC character sheet
2. Add a deployment
3. Advance to Level 1

**Expected behavior**
I would expect the "rank" 0/5 show 1/5, increasing with whatever rank is set. additionally I would expect the appropriate actions to be listed based on rank- or at least be in a compendium to add manually. 

**Screenshots**
![image](https://github.com/sw5e-foundry/sw5e/assets/117317204/ae9d565e-88d6-4e43-a615-fd176024788e)

**Foundry Information
Foundry version: [[Version]]
SW5e version: [[Version]]
OS: [[OS Information (Windows, macos, linux, etc)]]

**Additional context**

**Comment by Ikaguia** [comment]
If by `actions` you mean deployment features, all of them are under the aptly named `deployment features` compendium.
If you mean by actual actions you can take in combat, the only restricted actions are those that can only be done by the player at the helm (doesn't even need the pilot deployment). Also, all those actions are in the `starship actions` compendium.
Finally, deployments were missing a DataModel, and thus were unable to get advancements, such as those that automatically grant features on level up. I'm fixing this with my latest PR, so all that needs to be done is someone manually adding the advancement and choices, like this:
![image](https://github.com/sw5e-foundry/sw5e/assets/16377028/60a7712e-f0ed-4a93-a9e6-b2507d199227)

**Comment by Ikaguia** [comment]
@zeldafreak6245 can you comment on this?

**Comment by zeldafreak6245** [comment]
@Ikaguia What exactly did you want me to comment on?  If in regards to changes, I would say to remove the 0/5. Then Ideally having role-specific maneuvers / tactics options within the deployment itself.

**Comment by Ikaguia** [comment]
@zeldafreak6245 agreed on the 0/5 part.

> Then Ideally having role-specific maneuvers / tactics options within the deployment itself.
Do you mean as advancements, like the screenshot I posted above?

**Comment by zeldafreak6245** [comment]
@Ikaguia  Yes, but once selected it would only show the options they had picked, if that makes sense. So essentially when they gain a rank, they get options and pick whatever they want. Then after picking, only what they showed is still visible

**Comment by Ikaguia** [comment]
@zeldafreak6245 yeah, that is the behaviour 'Choose Items' advancements have. The code for this is already done, all that is needed is to make the advancements manually (like I did for pilot as an example).

**Comment by zeldafreak6245** [comment]
@Ikaguia I can attempt to help with that, but I am currently unsure how. Feel free to ping me on discord if you wanted to go over it- thanks!
